backend/internal/database/book.go

```diff
@@ -297,9 +297,129 @@ func (db *BookDatabase) IncrementBookCounter(ctx context.Context, id string) err
  */
 func (db *BookDatabase) FetchBookAnalytics(ctx context.Context, id string, dRange int) ([]int, error) {
 	var analytics []int
+
+	var firstYearStart int
+	var firstYearEnd int
+	var secondYearEnd int
+
+	firstYearStart, firstYearEnd, secondYearEnd = AnalyticsClicksIndices(dRange)
+
+	var query string = "SELECT ARRAY_CAT(clicks[$1:$2], clicks[1:$3]) " +
+		"FROM book_analytics WHERE id = $4"
```

Comment:
IMO remove AnalyticsClickIndices call, replace with `clicks[366-dRange+today:] || clicks[today-dRange:today]`. You can get the value for today from YearDay() in Go, no need to try and generate it in SQL. Also maybe we should rename dRange to numDays?